resources/pod.py

```diff
@@ -0,0 +1,158 @@
+"""
+Pod related functionalities and context info
+
+Each pod in the openshift cluster will have a corresponding pod object
+"""
+import logging
+import tempfile
+import yaml
+
+from ocs.ocp import OCP
+from ocs import defaults, kinds
+from ocsci.config import ENV_DATA
+from ocs.exceptions import CommandFailed
+
+from resources.base_resource import BaseOCSClass
+
+logger = logging.getLogger(__name__)
+
+
+class Pod(BaseOCSClass):
+    """
+    Handles per pod related context
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        kwargs:
+            Copy of ocs/defaults.py::<some pod> dictionary
+        """
+        self.pod_data = kwargs
+        super(Pod, self).__init__(
+            self.pod_data.get('apiVersion'), self.pod_data.get('kind'),
+            self.pod_data.get('metadata').get('namespace')
+        )
+
+        self.temp_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='POD_', delete=False
+        )
+        self._name = self.pod_data.get('metadata').get('name')
+        self._labels = self.get_labels()
+        self._roles = [self.pod_data.get('metadata').get('labels').get('app')]
+        self.ocp = OCP(
+            api_version=defaults.API_VERSION, kind=kinds.POD,
+            namespace=self.namespace
+        )
+        # TODO: get backend config !!
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @property
+    def roles(self):
+        return self._roles
+
+    @property
+    def labels(self):
+        return self._labels
+
+    def set_role(self, role):
+        """
+        Set a new role for this pod
+
+        Args:
+            role (str): New role to be assigned for this pod
+        """
+        self._roles.append(role)
+
+    def exec_cmd_on_pod(self, command):
+        """
+        Execute a command on a pod (e.g. oc rsh)
+
+        Args:
+            command (str): The command to execute on the given pod
+
+        Returns:
+            Munch Obj: This object represents a returned yaml file
+        """
+        rsh_cmd = f"rsh {self.name} "
```

Comment:
@shylesh can you please elaborate on that more?
any example I can use?

Reply:
@shylesh which pod exec will fail?

Reply:
@RazTamir @vasukulkarni 
It was on similar lines of https://github.com/red-hat-storage/ocs-ci/pull/156#discussion_r290978058, where pod name was drawn from yaml file which might fail. But since Raz said realoding does the magic of handling it then I am fine.